### PR TITLE
Theoretical background documentation

### PR DESCRIPTION
**Pull request by robbievanleeuwen**

### CODE REVIEW

File: sectionproperties/analysis/cross_section.py [hunk 1]
```diff
@@ -2525,26 +2526,60 @@ def get_stress(self):
         * *'sig_3'*: Minor principal stress :math:`\sigma_{3}` resulting from all actions
         * *'sig_vm'*: von Mises stress :math:`\sigma_{vM}` resulting from all actions
 
-        The following example returns the normal stress within a 150x90x12 UA section resulting
-        from an axial force of 10 kN::
+        The following example returns stresses for each material within a composite section, note
+        that a result is generated for each node in the mesh for all materials irrespective of
+        whethter the materials exists at that point or not.
```
Comment (connorferster) [inline sectionproperties/analysis/cross_section.py:2531]: Typo: whether

File: docs/source/rst/theory.rst [hunk 2]
```diff
@@ -4,12 +4,46 @@ Theoretical Background
...
+slender steel structures when lateral-torsional buckling is critical.
+
+Warping independent properties for basic cross-sections are relatively simple to calculate by hand.
+However accurate warping independent properties, even for the most basic cross-section, require
+solving several boundary value partial differential equations. This necessitates numerical methods
+in the calculation of these properties, which can be extended to arbitrary complex sections.
+
+This section of the documentation describes the theory and application of the finite element method
+to cross-sectional analysis used by *sectionproperties*. The goal of *sectionproperties* is to
+perform cross-sectional and stress analysis on arbitrary cross-sections, see below figure. In it's
```
Comment (connorferster) [inline docs/source/rst/theory.rst:21]: it's -> its

File: docs/source/rst/theory.rst [hunk 3]
```diff
@@ -18,37 +52,277 @@ Finite Element Preliminaries
 Element Type
 ^^^^^^^^^^^^
 
-element
+Quadratic six noded triangular elements were implemented in *sectionpropertes* in order to utilise
```
Comment (connorferster) [inline docs/source/rst/theory.rst:55]: Typo: sectionproperties

File: docs/source/rst/theory.rst [hunk 4]
```diff
@@ -330,43 +330,161 @@ Calculation of Cross-Section Properties
...
+  Z_{yy}^+ = \frac{I_{\overline{yy}}}{x_{max} - x_c} \\
+  Z_{yy}^- = \frac{I_{\overline{yy}}}{x_c - x_{min}} \\
+
+.. _label-theory-plastic-section-moduli:
 
 Plastic Section Moduli
 ^^^^^^^^^^^^^^^^^^^^^^
 
-moduli
+For a homogenous section, the plastic centroid can be determined by by finding the intersection of
```
Comment (connorferster) [inline docs/source/rst/theory.rst:431]: Typo: by by

File: docs/source/rst/theory.rst [hunk 5]
```diff
@@ -879,15 +879,38 @@ mohr's circle - to be completed...
...
+However, for composite sections with variable Poisson's ratios, this assumption does not hold.
+Pilkey [2] does mention that engineering materials often have very similar Poisson's ratios and
+therefore the difference can be negligible.
+
+.. note::
+  If the Poisson's ratio of two materials used in a composite analysis are vastly different, the
+  assumptions used in *sectionproperties* may not hold, see Chapter 5 & 6 of [2].
+
+For the warping and shear analysis of composite cross-sections, *sectionproperties* defines an area
+based effective Poisson's ratio that is used to calculate the relevant properties described above:
```
Comment (connorferster) [inline docs/source/rst/theory.rst:909]: Typo? based effective -> based on the effective

**Review by connorferster — APPROVED** [review]
Nicely written, Robbie! A big effort to put together!

**Comment by robbievanleeuwen** [comment]
Thanks for the review @connorferster, much appreciated!

**Pull request merged**